src/main/java/subway/domain/PathFinder.java

```diff
@@ -0,0 +1,47 @@
+package subway.domain;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+public class PathFinder {
+
+    private final List<Section> sections;
+
+    public PathFinder(final List<Section> sections) {
+        this.sections = sections;
+    }
+
+    public Path findShortestPath(final Station departure, final Station arrival) {
+        final Station namedDeparture = addNameToStation(departure);
+        final Station namedArrival = addNameToStation(arrival);
+        final GraphPath<Station, DefaultWeightedEdge> shortestPath = getDijkstraShortestPath().getPath(namedDeparture, namedArrival);
+        return new Path(shortestPath.getVertexList(), shortestPath.getWeight());
+    }
+
+    private Station addNameToStation(final Station station) {
+        final Optional<Station> result = sections.stream()
+                .flatMap(section -> Stream.of(section.getUpStation(), section.getDownStation()))
+                .filter(s -> s.equals(station))
+                .findFirst();
+
+        return result.orElseThrow(() -> new IllegalArgumentException("노선에 등록되지 않은 역입니다.: " + station.getId()));
+    }
+
+    private DijkstraShortestPath<Station, DefaultWeightedEdge> getDijkstraShortestPath() {
```

Comment:
직전 리뷰와 마찬가지 리뷰로, ShortestPathAlgorithm 인터페이스를 반환한다면 차후 알고리즘에 있어 확장성을 가져갈 수 있습니다.
```suggestion
    private ShortestPathAlgorithm<Station, DefaultWeightedEdge> getDijkstraShortestPath() {
```

Reply:
반영했습니다 😊